bin/blocks-plot

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python 
+
+from __future__ import division, print_function
+
+import dill
+import pickle
+import fnmatch
+import logging
+import readline
+import argparse
+import numpy 
+import pandas
+
+from six import iteritems
+from six.moves import input
+from collections import OrderedDict, defaultdict
+
+from blocks import config
+from blocks.utils import change_recursion_limit
+
+
+def log_channels(log):
+    """Return all channels contained in blocks.Log object
+    """
+    channels = set()
+    for t, channel, value in log:
+        channels.add(channel)
+    return channels
+
+def log_to_dataframe(log, channels=None, epochs_only=True):
+    """ Convert the content of a blocks.Log into a panda.DataFrame
+
+    Iterates over al log-records and creates a DataFrame. The time of
+    each log-record will be the row index. Each channels will create
+    a column in the DataFrame.
+
+    Parameters
+    ----------
+    channels: 
+        Which channels 
+
+    epochs_only: bool
+        If True, all; log-records which do not correspond to a full epoch w
+        be filtered out. Full epochs are identified by having a 'epoch-took'
+        record, or by having a time of 0.
+
+    Returns
+    -------
+    DataFrame
+    """
+    channels = defaultdict(dict)
+    for t, channel, value in log:
+        # TODO: ensure value is something we actually want to keep.
+        # We probably have to discard all non-scalars as they can not
+        # be stored in a DataFrame.
+        # TODO: ensure that the channels have "homogeneous" data types
+        channels[channel] [t] = value
+
+    df = pandas.DataFrame(channels)
+
+    if epochs_only:
```

Comment:
I think your should consider `log.status._epoch_ends`. This is a list of end iterations of all epochs.